lxd/db/cluster/secrets.go

```diff
@@ -0,0 +1,135 @@
+package cluster
+
+import (
+	"crypto/rand"
+	"database/sql/driver"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/canonical/lxd/lxd/db/query"
+)
+
+// authSecretLength defines the length of an AuthSecretValue.
+const authSecretLength = 64
+
+// AuthSecretValue is used to byte slice of length authSecretLength in a text column as a base64 encoded string.
+type AuthSecretValue []byte
+
+// newAuthSecretValue returns a AuthSecretValue, read from [rand.Reader].
+func newAuthSecretValue() AuthSecretValue {
+	secretValue := make([]byte, authSecretLength)
+	_, _ = rand.Read(secretValue)
```

Comment:
Does this need seeding or is that not required these days in Go?

Reply:
The `math/rand` package can be seeded because it uses a psuedorandom number generator thats defined in Go, but I don't think `crypto/rand` can be seeded because it calls OS APIs ([getrandom](https://manpages.debian.org/unstable/manpages-dev/getrandom.2.en.html), falling back to reading from `/dev/urandom`).

Reply:
ah yes ofcourse, thanks